cpp/src/distance/detail/distance.cuh

```diff
@@ -298,11 +298,14 @@ void distance_impl(raft::resources const& handle,
 {
   cudaStream_t stream = raft::resource::get_cuda_stream(handle);
 
+  rmm::device_uvector<DataT> y_temp(n * k, stream);
+  raft::copy(y_temp.data(), y, n * k, stream);
```

Comment:
Isn't this copying the whole input matrix? This can be huge! I've been trying to understand the tiling logic, but looking at the hellinger distance op, it seems like that is just a product of sqrt(x) and sqrt(y), so is it possible to do the sqrt and revert for y in place?
On the other hand, if you do require to not alter the y matrix, is there a way you can embed the sqrt in the hellinger op itself?

Reply:
Thanks for the review Tarang! I managed to do it in-place with some pointer arithmetic.